### DISCUSSION TITLE
Whitelabeling Passwordless Email Login

### DISCUSSION THREAD

**Discussion by Animalmix55** [Web3Auth/Web3Auth#1294] [Help]
Hello!

I am leveraging @web3auth/core using a few different login methods. I have managed to whitelabel the OpenLoginAdapter in the popups that appear during the login flow for my custom JWT, email, and Google logins, but for the email login, the email sent to the user does not obey the theming specified nor does in have the app logo.

Can this email be customized without self hosting, if so, how? As it seems to be separate from the customization options on the OpenLoginAdapter.

**Comment by arch1995** [comment]
Hi @Animalmix55 

We have added support to Whitelabel passwordless emails using the same customization options on the OpenloginAdapter. 

If this doesn't work out of the box, please update the SDKs to the latest version.

**Reply by Animalmix55** [reply]
Hi @arch1995 

I went ahead and updated to version 4.4.0 of all packages and attempted a passwordless login, this is the email I received:

```
    "@web3auth/base": "^4.4.0",
    "@web3auth/coinbase-adapter": "^4.4.0",
    "@web3auth/core": "^4.4.0",
    "@web3auth/ethereum-provider": "^4.4.0",
    "@web3auth/metamask-adapter": "^4.4.0",
    "@web3auth/openlogin-adapter": "^4.4.0",
    "@web3auth/ui": "^4.4.0",
    "@web3auth/wallet-connect-v1-adapter": "^4.4.0",
```

![image](https://user-images.githubusercontent.com/4249734/217443386-753187f9-711e-4c5a-a510-c6e896d47a5a.png)

This is my `OpenLoginAdapter` configuration:

```
adapterSettings: {
                network,
                clientId,
                uxMode: 'popup',
                whiteLabel: {
                    defaultLanguage: 'en',
                    name: '<URL>',
                    logoLight:  '<URL>',
                    logoDark: '<URL>',
                    theme: {
                        primary: '<PRIMARY_HEX>',
                    },
                    dark: true,
                },
                loginConfig: {
                    ...baseLoginConfig,
                    Google: {
                        name: 'Google',
                        verifier: <VERIFIER_ID>,
                        typeOfLogin: 'google',
                        clientId: googleClientId,
                    },
                },
            },
```

The primary color passed is a purplish hue, and the logo passed is nowhere present on the email above. Maybe there's some miscommunication here?

**Reply by arch1995** [reply]
Could you try hard-refreshing the page when you are on the Email Passwordless mail sent confirmation screen (`auth.openlogin.com/authorize`) 

I feel there can be a service worker issue on this page.

**Reply by Animalmix55** [reply]
So, I tried to give that a go this morning, but I have a new issue with initiating email login (this occurs before the email is sent on the requesting screen)

![image](https://user-images.githubusercontent.com/4249734/217593454-cc37d61a-99f3-4b87-a888-92c219e4274e.png)

**Reply by arch1995** [reply]
I see, in the request `whiteLabel.name` is not a valid string. It should not consist of characters like `:`

Also, We recommend using PNG or JPG for logos, because SVG won't show up on Gmail

**Reply by Animalmix55** [reply]
Awesome, just updated the `name` and `logoLight`/`logoDark` attributes and we are off to the races! Thank you!

**Reply by Haavi97** [reply]
Thanks!
This should be added to the documentation. I also created a post in the community forum about it:
https://web3auth.io/community/t/logo-format-must-be-png-or-jpg-to-appear-in-the-verification-email-when-using-email-passwordless-whitelabel-add-to-documentation/6855
